src/runtime/auto-prompt-manager-test.js

```diff
@@ -1526,24 +1548,303 @@ describes.realWin('AutoPromptManager', (env) => {
       expect(autoPromptManager.promptDisplayed_).to.equal(
         'TYPE_REGISTRATION_WALL'
       );
-      await verifyOnCancelStores('contribution,TYPE_REGISTRATION_WALL');
+      await verifyOnCancelStores(
+        storageMock,
+        actionFlowSpy,
+        'contribution,TYPE_REGISTRATION_WALL'
+      );
     });
+  });
 
-    async function verifyOnCancelStores(setValue) {
-      storageMock
-        .expects('set')
-        .withExactArgs(
-          StorageKeys.DISMISSED_PROMPTS,
-          setValue,
-          /* useLocalStorage */ true
-        )
-        .resolves(null)
+  describe('Contribution Flows with Audience Actions and Experiments', () => {
```

Comment:
One comment on these tests: these are almost entirely redundant of the original `Contribution Flows with Audience Actions` test, with some slight modifications -- primarily this suite checks for the main case (where the survey prompt should be triggered before the contribution prompt when both are valid to show), and the other tests assert the behavior is the same as with the experiment disabled in all other cases. 

There may be a more concise way to test this with parameters, but I think there is an argument for keeping an explicit suite for Experiments, which can easily be added to / ripped out as we launch/turn down experiments.